src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -393,40 +360,80 @@ static BuildEnvironmentHelperSingleton()
     }
 
     /// <summary>
+    /// Enum which defines which environment / mode MSBuild is currently running.
+    /// </summary>
+    internal enum BuildEnvironmentMode
+    {
+        /// <summary>
+        /// Running from Visual Studio directly or from MSBuild installed under an instance of Visual Studio.
+        /// Toolsets and extensions will be loaded from the Visual Studio instance.
+        /// </summary>
+        VisualStudio,
+
+        /// <summary>
+        /// Running in a standalone toolset mode. All toolsets and extensions paths are relative to the app 
+        /// running and not dependent on Visual Studio. (e.g. dotnet CLI, open source clone of our repo)
+        /// </summary>
+        Standalone,
+
+        /// <summary>
+        /// Running without any defined toolsets. Most functionality limited. Likely will not be able to
+        /// build or evaluate a project. (e.g. reference to Microsoft.*.dll without a toolset definition
```

Comment:
Maybe mention the running-out-of-a-package example, since that would probably be the usual case.